### DISCUSSION TITLE
getIterator Scan and ScanFilter using decimal values

### DISCUSSION THREAD

**Discussion by trrisner** [aws/aws-sdk-php#2795] [Q&A]
When working with the below code, I get an error because the $lat1 is not a string. 
I can convert to a string with (string)$lat1 ...but this doesn't work for decimal values. 
I'm trying to find all values between the lat/lon of $lat1,$lat2,$lon1, and $lon2   

             $iterator2=$client2->getIterator("Scan",array(
                        "TableName"=>"keyfile",
                        "ScanFilter"=>array(
                                "latitude"=>array(
                                        "AttributeValueList"=>array(array("N"=>$lat1)),
                                        "ComparisonOperator"=>"GT"
                                                 ),
                                "latitude"=>array(
                                        "AttributeValueList"=>array(array("N"=>$lat2)),
                                        "ComparisonOperator"=>"LT"
                                                 ),
                                "longitude"=>array(
                                        "AttributeValueList"=>array(array("N"=>$lon1)),
                                        "ComparisonOperator"=>"GT"
                                                 ),
                                "longitude"=>array(
                                        "AttributeValueList"=>array(array("N"=>$lon2)),
                                        "ComparisonOperator"=>"LT"
                                                 )
                                            )
                                                        ));

**Comment by yenfryherrerafeliz** [comment]
Hi @trrisner, sorry to hear about your issues. Can you please provide a full sample code where we can see what you are importing, which SDK client you are using, etc.?

Thanks!

**Reply by trrisner** [reply]
I'm using the most recent release of aws-sdk-php on the most recent AmazonLinux 2023 release. 
PHP 8.2.9

<php

require 'vendor/autoload.php'; 
use Aws\DynamoDb\DynamoDbClient; 
$client=DynamoDbClient::factory(array(
"credentials"=>array('key'=>$AWS_KEY,'secret'=>$AWS_SECRET),
"region"=>$OPERATIONALREGION));


         $iterator2=$client->getIterator("Scan",array(
                    "TableName"=>"keyfile",
                    "ScanFilter"=>array(
                            "latitude"=>array(
                                    "AttributeValueList"=>array(array("N"=>$lat1)),
                                    "ComparisonOperator"=>"GT"
                                             ),
                            "latitude"=>array(
                                    "AttributeValueList"=>array(array("N"=>$lat2)),
                                    "ComparisonOperator"=>"LT"
                                             ),
                            "longitude"=>array(
                                    "AttributeValueList"=>array(array("N"=>$lon1)),
                                    "ComparisonOperator"=>"GT"
                                             ),
                            "longitude"=>array(
                                    "AttributeValueList"=>array(array("N"=>$lon2)),
                                    "ComparisonOperator"=>"LT"
                                             )
                                        )
                                                    ));


===========================================================================


ERROR is showing up as $lat1, $lat2, $lon1, and $lon2 not  being a string. 
If I declare them a string at any point to correct that error, you get an error saying it's a decimal so it cannot be a string, which makes sense. In previous versions, $lat1, $lat2, $lon1, and $lon2 didn't have to be a string and actually had to be an integer or float since each are defined as an "N". Just curious if this is my error or an unknown change that was made?


=============================================================================

I did find a solution. 
If you round $lat1, $lat2, $lon1, and $lon2 to the nearest whole number, then declare them a string, you get the data you need, just with many more data points because your boundaries are wider. I then looped through $iterator2 and used my original boundaries in an if/then loop before declaring the output array needed. 

So, I coded around this to find a solution, but still wanted it to be known. 

Thanks for all that has been done with aws-sdk-php!
